cellmlmanip/units.py

```diff
@@ -385,11 +499,29 @@ def traverse(self, expr: sympy.Expr):
                 return out
 
             logger.warning('Piecewise args do not have the same unit.')
-            return None
+            raise InputArgumentsInvalidUnitsError('%s' % expr)
+
+        elif expr.is_Relational:
+            # following discussion with Michael we decided that since a
+            # variable in cellml can never have a boolean value then
+            # we should not encounter expression that return booleans
+            # but cellml spec says these should have same arguments so
+            # log this
+            if not self._check_unit_of_quantities_equal(quantity_per_arg):
+                logger.warning('Relational args do not have the same unit: %s', expr)
+            logger.critical('Boolean return: %s', expr)
+            raise BooleanUnitsError('%s' % expr)
+
+        elif expr.is_Boolean:
+            # following discussion with Michael we decided that since a
+            # variable in cellml can never have a boolean value then
+            # we should not encounter expression that return booleans
+            raise BooleanUnitsError('%s' % expr)
 
         elif expr.is_Function:
             # List of functions that have been checked
-            if str(expr.func) not in ['cos', 'acos', 'exp', 'floor', 'log', 'Abs', 'tanh']:
+            if str(expr.func) not in ['cos', 'acos', 'exp', 'floor', 'log', 'Abs', 'tanh', 'ceiling',
```

Comment:
Shouldn't that be Pow? I tried pow with lowercase the other day in the interpreter and it didn't work but Pow did